ravendb/resource_ravendb_server.go

```diff
@@ -317,45 +329,103 @@ func parseData(d *schema.ResourceData) (ServerConfig, error) {
 
 	urlSet := d.Get("url").(*schema.Set).List()
 	for _, v := range urlSet {
-		value := v.(map[string]interface{})
+		value := v.(map[string]interface{}) // After this stage the ports will get zero values.
 		list := value["list"].([]interface{})
 		sc.Url.List = make([]string, len(list))
 		for i, url := range list {
 			sc.Url.List[i] = url.(string)
 		}
-		if httpPort, ok := value["http_port"]; ok {
-			sc.Url.HttpPort = httpPort.(int)
-		} else {
+		if value["http_port"] == 0 {
 			sc.Url.HttpPort = DEFAULT_SECURE_RAVENDB_HTTP_PORT
 			if sc.Unsecured {
 				sc.Url.HttpPort = DEFAULT_USECURED_RAVENDB_HTTP_PORT
 			}
-		}
-
-		if tcpPort, ok := value["tcp_port"]; ok {
-			sc.Url.TcpPort = tcpPort.(int)
 		} else {
+			sc.Url.HttpPort = value["http_port"].(int)
+
+		}
+		if value["tcp_port"] == 0 {
 			sc.Url.TcpPort = DEFAULT_SECURE_RAVENDB_TCP_PORT
 			if sc.Unsecured {
 				sc.Url.TcpPort = DEFAULT_UNSECURED_RAVENDB_TCP_PORT
 			}
+		} else {
+			sc.Url.TcpPort = value["tcp_port"].(int)
 		}
 	}
 
-	if sc.ClusterCertificate != nil && sc.Unsecured == true {
-		return sc, fmt.Errorf("expected unsecure to be ture. certificate should be added when using secure mode")
+	if sc.ClusterSetupZip != nil && sc.Unsecured == true {
+		return sc, fmt.Errorf("expected unsecure to be ture. Certificate should be added when using secure mode")
 	}
 
 	return sc, nil
 }
-func allZero(s []byte) bool {
-	for _, v := range s {
-		if v != 0 {
-			return false
+
+func OpenZipFile(sc ServerConfig, path string) (map[string]*CertificateHolder, error) {
+	var split []string
+	var zipStruct *CertificateHolder
+
+	zipReader, err := zip.OpenReader(path)
+	if err != nil {
+		return nil, err
+	}
+	defer zipReader.Close()
+
+	var rc io.ReadCloser // to avoid defer inside the loop
```

Comment:
You can kill this variable, it isn't used

Reply:
it is used in line 392. Am I missing something?

Reply:
You are passing it to a method, but not using this for any reason